### PR TITLE
Handle OperationCancelled when peer disconnects during DAO fork check

### PR DESCRIPTION
**Pull request by gsalgado**
Closes: #1234

### CODE REVIEW

**Comment by gsalgado** [comment]
Alternative solutions I considered:

In order to perform the dao check in Peer._run() we'd need to run it in the background
and use a notification mechanism other than an Event so that we can tell the PeerPool
whether the check was successful or not. That is very error prone, though, because a
callsite may very easily forget to wait on that condition and start to use the peer straight
away. And we have no way of putting checks on peer methods to avoid them being called before
the DAO fork check has completed because we need to use those methods in order to perform
the DAO fork check.

Also thought about having the DAO fork check in the handshake() func, but then we'd have to
duplicate some code from _run() to read messages, and it'd also have to return the messages
received during the DAO check so that the PeerPool can re-add them to its subscribers. In
fact it'd require a lot of changes to make that work because the peer is not running at that
time and the new ExchangeManager (which is used during the DAO check) stuff requires the
peer to be running.

File: p2p/peer.py [hunk 1]
```diff
@@ -956,10 +963,7 @@ def get_peers(self, min_td: int) -> List[BasePeer]:
                 for line in peer.get_extra_stats():
                     self.logger.debug("    %s", line)
             self.logger.debug("== End peer details == ")
-            try:
-                await self.sleep(self._report_interval)
-            except OperationCancelled:
-                break
```
Comment (gsalgado) [inline p2p/peer.py]: This was no longer necessary as this coroutine now runs via `BaseService.run_task()`

**Comment by pipermerriam** [comment]
Alright, seeing this I want to take a stab at an alternate idea.

**Comment by pipermerriam** [comment]
Alright, it's really messy but I think it actually attacks the root of the problem as well as doing some cleanup migrating things that still live in the `p2p` module to be under `trinity`.

see https://github.com/ethereum/py-evm/pull/1260 as my proposed alternative to this PR

**Comment by carver** [comment]
~~#1260~~
#1259 I think

**Comment by pipermerriam** [comment]
#1265 is now the correct link.

**Pull request closed without merging**